oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/editor/FulltextDocumentMaker.java

```diff
@@ -336,6 +338,32 @@ private boolean indexProperty(String path,
         return dirty;
     }
 
+    private String getValueAsTruncatedString(Object val) {
+        String value = val.toString();
+        return value.length() <= 128 ? value
+            : value.substring(0, 128) + " ... [" + value.hashCode() + "]";
```

Comment:
This is going to fail with StringIndexOutOfBoundsException if the value is smaller than 128.

Reply:
I don't see how? The first condition in the ternary returns the value if it's smaller than 128?

Reply:
Ok, it's correct, I got confused by the use of the ternary operator.